lib/repo_tools.pm

```diff
@@ -81,7 +81,10 @@ This repository *is* mandatory.
 
 =cut
 sub add_qa_head_repo {
-    zypper_ar(get_required_var('QA_HEAD_REPO'), name => 'qa-head', no_gpg_check => is_sle("<12") ? 0 : 1);
+    my (%args)   = @_;
+    my $priority = $args{priority};
```

Comment:
What will happen if priority is not in argument ?  I suppose default value should be set if  argument is not present.

Reply:
exactly. It is handled in zypper_ar if I get it right

Reply:
But, if this is OK, I would still ask @pevik  to take a look and perhaps propose this change in a separate PR, as ltp does use this function. wdyt?

Reply:
But will not `$args{priority}` fail if it is not existent?  Maybe this could be better:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/31b082a63fd328072a03b14e86f37f4919a78f27/lib/opensusebasetest.pm#L582

with 99 as default.

Reply:
> But will not `$args{priority}` fail if it is not existent? Maybe this could be better:
> 
> https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/31b082a63fd328072a03b14e86f37f4919a78f27/lib/opensusebasetest.pm#L582
> 
> with 99 as default.

sure, this makes perfect sens

Reply:
> But will not `$args{priority}` fail if it is not existent? Maybe this could be better:
> 
> https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/31b082a63fd328072a03b14e86f37f4919a78f27/lib/opensusebasetest.pm#L582
> 
> with 99 as default.

pushed with // 99

Reply:
> But, if this is OK, I would still ask @pevik to take a look and perhaps propose this change in a separate PR, as ltp does use this function. wdyt?

Yes, change to repo_tools function should be clearly visible in git commit history and not hidden inside blktests PR.

Reply:
> > But, if this is OK, I would still ask @pevik to take a look and perhaps propose this change in a separate PR, as ltp does use this function. wdyt?
> 
> Yes, change to repo_tools function should be clearly visible in git commit history and not hidden inside blktests PR.

Of course I kept git commits separate as those are independent changes, so git log would be clean anyway. Given Petr V. gave his +1, I would keep both git commit in the same PR now.